app/src/pages/admin/pluginsPage/pluginsFilter/pluginsFilter.jsx

```diff
@@ -14,49 +14,49 @@
  * limitations under the License.
  */
 
-import React, { Component, Fragment } from 'react';
+import React, { Fragment } from 'react';
 import PropTypes from 'prop-types';
+import { useTracking } from 'react-tracking';
 import classNames from 'classnames/bind';
 import { getPluginsFilter } from 'common/constants/pluginsFilter';
+import { PLUGINS_PAGE_EVENTS } from 'components/main/analytics/events';
+import { PLUGIN_LABELS } from 'common/constants/pluginsGroupTypes';
 import styles from './pluginsFilter.scss';
 
 const cx = classNames.bind(styles);
 
-export class PluginsFilter extends Component {
-  static propTypes = {
-    filterItems: PropTypes.array.isRequired,
-    onFilterChange: PropTypes.func.isRequired,
-    activeItem: PropTypes.string.isRequired,
-  };
-
-  getFilterItems = () => getPluginsFilter(this.props.filterItems);
+export const PluginsFilter = ({ filterItems, onFilterChange, activeItem }) => {
+  const { trackEvent } = useTracking();
+  const getFilterItems = () => getPluginsFilter(filterItems);
 
-  changeFilterItem = (e) => {
+  const changeFilterItem = (e) => {
     e.preventDefault();
-    this.props.onFilterChange(e.currentTarget.id);
-  };
-
-  generateItems = () => {
-    const { activeItem } = this.props;
-
-    return (
-      <ul className={cx('plugins-filter-list')}>
-        {this.getFilterItems().map((item) => (
-          <li key={item.value} className={cx('plugins-filter-item')}>
-            <button
-              className={cx('plugins-filter-button', { active: activeItem === item.value })}
-              onClick={this.changeFilterItem}
-              id={item.value}
-            >
-              {item.label}
-            </button>
-          </li>
-        ))}
-      </ul>
-    );
+    const { id } = e.currentTarget;
+    trackEvent(PLUGINS_PAGE_EVENTS.navigatedInPluginsFilterList(PLUGIN_LABELS[id]));
+    onFilterChange(id);
   };
 
-  render() {
-    return <Fragment>{this.generateItems()}</Fragment>;
-  }
-}
+  const generateItems = () => (
+    <ul className={cx('plugins-filter-list')}>
+      {getFilterItems().map((item) => (
+        <li key={item.value} className={cx('plugins-filter-item')}>
+          <button
+            className={cx('plugins-filter-button', { active: activeItem === item.value })}
+            onClick={changeFilterItem}
+            id={item.value}
+          >
+            {item.label}
+          </button>
+        </li>
+      ))}
+    </ul>
+  );
+
+  return <Fragment>{generateItems()}</Fragment>;
```

Comment:
I suppose we don't need a `Fragment` here.
Let's also remove `generateItems` function and return JSX directly.